app/src/main/java/com/kms/cura/view/activity/DoctorProfileViewActivity.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+
+    class ViewHolder {
+        private TextView txtFacilityName, txtFacilityAddress, txtFacilityPhoneNumber;
+        private LinearLayout layoutFacilityTime;
+        private View root = null;
+        private FacilityEntity entity;
+
+        public ViewHolder(View root, FacilityEntity entity) {
+            this.root = root;
+            this.entity = entity;
+        }
+
+        public TextView loadText(String src, int id) {
+            TextView textView = (TextView) root.findViewById(id);
+            textView.setText(src);
+            return textView;
+        }
+
+        public void loadFacility() {
+            txtFacilityName = loadText(entity.getName(), R.id.txtDoctorFacilityName);
+            txtFacilityAddress = loadText(DataUtils.showUnicode(entity.getAddress()), R.id.txtDoctorFacilityAddress);
+            txtFacilityPhoneNumber = loadText(entity.getPhone(), R.id.txtDoctorFacilityPhoneNumber);
+        }
+
+
+    }
+
+
+    private List<HashMap<String, OpeningHour>> seedData() {
+        List<HashMap<String, OpeningHour>> list = new ArrayList<>();
+        for (int i = 0; i < 7; ++i) {
+            HashMap<String, OpeningHour> workingtime = new HashMap<>();
+            workingtime.put("A", new OpeningHour(DayOfTheWeek.getDayOfTheWeek(i), new Time(3, 3, 3), new Time(4, 4, 4)));
+            workingtime.put("B", new OpeningHour(DayOfTheWeek.getDayOfTheWeek(i), new Time(3, 3, 3), new Time(4, 4, 4)));
+            list.add(workingtime);
+        }
+        return list;
+    }
+
+    private void setListViewHeight(ExpandableListView listView,
+                                   int group) {
+        ExpandableListAdapter listAdapter = (ExpandableListAdapter) listView.getExpandableListAdapter();
+        int totalHeight = 0;
+        int desiredWidth = View.MeasureSpec.makeMeasureSpec(listView.getWidth(),
+                View.MeasureSpec.EXACTLY);
+        for (int i = 0; i < listAdapter.getGroupCount(); i++) {
+            View groupItem = listAdapter.getGroupView(i, false, null, listView);
+            groupItem.measure(desiredWidth, View.MeasureSpec.UNSPECIFIED);
+
+            totalHeight += groupItem.getMeasuredHeight();
+
+            if (((listView.isGroupExpanded(i)) && (i != group))
+                    || ((!listView.isGroupExpanded(i)) && (i == group))) {
+                for (int j = 0; j < listAdapter.getChildrenCount(i); j++) {
+                    View listItem = listAdapter.getChildView(i, j, false, null,
+                            listView);
+                    listItem.measure(desiredWidth, View.MeasureSpec.UNSPECIFIED);
+
+                    totalHeight += listItem.getMeasuredHeight();
+
+                }
+            }
+        }
+
+        ViewGroup.LayoutParams params = listView.getLayoutParams();
+        int height = totalHeight
+                + (listView.getDividerHeight() * (listAdapter.getGroupCount() - 1));
+        if (height < 10)
+            height = 200;
```

Comment:
This method is too long and contains logic of getting height too. Should extract code from line 181 to 208 to another method